express/scripts/utils/icons.js

```diff
@@ -193,3 +193,55 @@ export async function fixIcons(el = document) {
     }
   });
 }
+
+export function decorateSocialIcons(block) {
+  block.querySelectorAll(':scope a').forEach(($a) => {
+    const urlObject = new URL($a.href);
+
+    if (urlObject.hash === '#embed-video') return;
+    if ($a.closest('.block')?.dataset.blockName === 'embed') return;
+
+    if ($a.href === $a.textContent.trim()) {
+      let icon = '';
+      if (urlObject.hostname === 'www.instagram.com') {
+        icon = 'instagram';
+      }
+      if (urlObject.hostname === 'www.twitter.com') {
+        icon = 'twitter';
+      }
+      if (urlObject.hostname.split('.')[1] === 'pinterest') {
+        icon = 'pinterest';
+      }
+      if (urlObject.hostname.split('.')[1] === 'facebook') {
+        icon = 'facebook';
+      }
+      if (urlObject.hostname === 'www.linkedin.com') {
+        icon = 'linkedin';
+      }
+      if (urlObject.hostname === 'www.youtube.com') {
+        icon = 'youtube';
+      }
+      if (urlObject.hostname === 'www.tiktok.com') {
+        icon = 'tiktok';
+      }
+      const $parent = $a.parentElement;
+      if (!icon && $parent.previousElementSibling && $parent.previousElementSibling.classList.contains('social-links')) {
+        icon = 'globe';
+      }
+
+      if (icon) {
+        $a.textContent = '';
+        const $icon = getIconElementDeprecated(icon, 22);
+        $icon.classList.add('social', 'is-visible');
+        $a.appendChild($icon);
+        $a.classList.add('social-link');
+        if ($parent.previousElementSibling && $parent.previousElementSibling.classList.contains('social-links')) {
+          $parent.previousElementSibling.appendChild($a);
+          $parent.remove();
+        } else {
+          $parent.classList.add('social-links');
+        }
+      }
+    }
+  });
+}
```

Comment:
Small nit: I didn't test this, but a switch case might be cleaner. 

```suggestion
export function decorateSocialIcons(block) {
  block.querySelectorAll(':scope a').forEach(($a) => {
    const urlObject = new URL($a.href);

    if (urlObject.hash === '#embed-video') return;
    if ($a.closest('.block')?.dataset.blockName === 'embed') return;

    if ($a.href === $a.textContent.trim()) {
      let icon = '';
      switch (urlObject.hostname) {
        case 'www.instagram.com':
          icon = 'instagram';
          break;
        case 'www.twitter.com':
          icon = 'twitter';
          break;
        case 'www.linkedin.com':
          icon = 'linkedin';
          break;
        case 'www.youtube.com':
          icon = 'youtube';
          break;
        case 'www.tiktok.com':
          icon = 'tiktok';
          break;
        default:
          const secondLevelDomain = urlObject.hostname.split('.')[1];
          if (secondLevelDomain === 'pinterest') {
            icon = 'pinterest';
          } else if (secondLevelDomain === 'facebook') {
            icon = 'facebook';
          }
      }

      const $parent = $a.parentElement;
      if (!icon && $parent.previousElementSibling && $parent.previousElementSibling.classList.contains('social-links')) {
        icon = 'globe';
      }

      if (icon) {
        $a.textContent = '';
        const $icon = getIconElementDeprecated(icon, 22);
        $icon.classList.add('social', 'is-visible');
        $a.appendChild($icon);
        $a.classList.add('social-link');
        if ($parent.previousElementSibling && $parent.previousElementSibling.classList.contains('social-links')) {
          $parent.previousElementSibling.appendChild($a);
          $parent.remove();
        } else {
          $parent.classList.add('social-links');
        }
      }
    }
  });
}
```

Reply:
good idea!
I think it can sometimes bite us in the butt to modify methods that have been copy pasted over,but in this case you're right that it probably won't be getting modified at all, so makes sense to make it better.